packages/wrangler/src/pages/build-env.ts

```diff
@@ -9,17 +12,37 @@ import type {
 export type PagesBuildEnvArgs = StrictYargsOptionsToInterface<typeof Options>;
 
 export function Options(yargs: CommonYargsArgv) {
-	return yargs;
+	return yargs.positional("projectDir", {
+		type: "string",
+		description: "The location of the Pages project",
+	});
 }
 
 export const Handler = async (args: PagesBuildEnvArgs) => {
-	const config = readConfig(undefined, {
+	if (!args.projectDir) {
+		throw new FatalError("No Pages project location specified");
+	}
+	const configPath = path.resolve(args.projectDir, "wrangler.toml");
+	// Fail early if the config file doesn't exist
+	try {
+		await stat(configPath);
+	} catch {
+		throw new FatalError(
+			"No Pages config file found",
+			EXIT_CODE_NO_CONFIG_FOUND
+		);
+	}
+	const config = readConfig(path.resolve(args.projectDir, "wrangler.toml"), {
 		...args,
 		// eslint-disable-next-line turbo/no-undeclared-env-vars
 		env: process.env.PAGES_ENVIRONMENT,
 	});
+	// Fail if the config file exists but isn't valid for Pages
 	if (!config.pages_build_output_dir) {
-		throw new FatalError("No Pages config file found");
+		throw new FatalError(
+			"No Pages config file found",
+			EXIT_CODE_NO_CONFIG_FOUND
```

Comment:
I'm assuming we need to exit with this exit code here? Cause technically we did find a `wrangler.toml` file, just it doesn't have the mandatory `pages_build_output_dir` Pages field. IMO the err message should reflect that, but not sure if we can exit with the same exit code but a diff err message than on line https://github.com/cloudflare/workers-sdk/pull/5327/files?diff=unified&w=1#diff-f0b20ef6f77561affbcad6b9afae83c2b49211c0e1d7af12a81abf191b72bbafR31 ?

Reply:
Yeah it feels a bit weird to exit with "no config found".
From the user's point of view they will see that error and get confused because they will see that there is a wrangler.toml and maybe not realise that it is just malformed.

Reply:
indeed, that was my thinking as well

Reply:
Added a new exit code

Reply:
I guess this will prolly need to be persisted in the BE work as well, so just calling that out

Reply:
Yep, I'll add a task